src/components/checkoutComponent/checkout.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useParams, useHistory } from 'react-router';
+import { fetchJob } from '../../store/job/jobActions';
+import { StoreState } from '../../store/store';
+import {
+  Table,
+  TableBody,
+  TableContainer,
+  TableHead,
+  TableCell,
+  Paper,
+  TableRow,
+  TextField,
+} from '@material-ui/core';
+import StripeCheckout from 'react-stripe-checkout';
+import axios from 'axios';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { JobAttributes } from '../../store/job/jobInterface';
+import { validate } from '../validation';
+
+interface RouteParams {
+  id: string;
+}
+
+const Checkout: React.FC = () => {
+  const dispatch = useDispatch();
+  const { user, job } = useSelector((state: StoreState) => state);
+  const { id } = useParams<RouteParams>();
+  const [price, setPrice] = useState('');
+  useEffect(() => {
+    const getJob = () => {
+      return new Promise(res => {
+        res(dispatch(fetchJob(id)));
+      }).then((data: JobAttributes) => {
+        setPrice(String(data.price));
+      });
+    };
+    getJob();
```

Comment:
fetchJob already returns a promise. you can call fetchJob then setPrice without all of the new Promise stuff.

Reply:
setPrice has to be done async using await with fetchJob does not perform setPrice afterwards, both ways have been tried

Reply:
why does price even need to be set to local state in this scenario?  why would anyone have the ability to change the price on the checkout page?